apps/src/templates/teacherDashboard/AssignmentVersionSelector.jsx

```diff
@@ -30,12 +30,54 @@ const styles = {
   }
 };
 
+/**
+ * Given an array of versions, return that array with the same versions, plus
+ * isRecommended and isSelected properties set on the recommented and selected version(s).
+ * @param {Array<AssignmentVersionShape>} versions
+ * @param {String} locale. User's current locale.
+ * @param {String} selectedVersionYear. Currently selected version year. Optional.
+ */
+export const setRecommendedAndSelectedVersions = (
```

Comment:
this would be a great function to unit test, unless it's inner workings are already getting covered another way

Reply:
yeah, you're right. the internals are tested by [this test](https://github.com/code-dot-org/code-dot-org/blob/f257bfde27941cbd350dd89ea83d319dfb4eb7aa/apps/test/unit/code-studio/components/progress/ScriptOverviewHeaderTest.js#L228-L272) that i just added, but this method should have it's own tests for clarity/future use. will add!

Reply:
Sorry, you are right -- I was mislead by the name `passes properly-formatted versions to AssignmentVersionSelector` but in reality it does more and is effectively testing the internal functionality of `setRecommendedAndSelectedVersions` already. Per Paul's lecture, adding unit test coverage on `setRecommendedAndSelectedVersions` would still help because a failure would tell you exactly where the error was, but now that I understand what it covers I'd say you've got it pretty well covered already and a unit test on `setRecommendedAndSelectedVersions` would be nice-to-have for extra credit. :-)

Reply:
Ah, sorry again, now I see that the existing test is really only testing one set of inputs for `setRecommendedAndSelectedVersions`. so, please do proceed with unit tests for `setRecommendedAndSelectedVersions`!